backend_plugin/src/main/evaluators/AddImportEvaluator.java

```diff
@@ -0,0 +1,104 @@
+package main.evaluators;
+
+import java.util.Iterator;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.DocumentEvent;
+import org.eclipse.jface.text.source.Annotation;
+import org.eclipse.jface.text.source.IAnnotationModel;
+
+public class AddImportEvaluator extends FeatureEvaluator {
+	
+	private boolean unresolvedVariablesExist;
+	private DocumentEvent lastDocumentEvent;
+	
+	/**
+	 * Construct an AddImportEvaluator
+	 */
+	public AddImportEvaluator() {
+		this.featureID = "addImportStatementsSuggestion";
+		this.lastDocumentEvent = null;
+		this.unresolvedVariablesExist = false;
+	}
+
+	/**
+	 * Tracks the user's changes, and determines whether they are fixing an
+	 * "unresolved type" issue by manually typing the import statement
+	 * @return boolean true if the user is manually adding an import statement
+	 * 					when unresolved variables exist; false otherwise
+	 */
+	@Override
+	public boolean evaluateDocumentChanges(DocumentEvent docEvent) {
```

Comment:
I'm a little worried about how this evaluator works. It seems to wait for a new annotation to pop up that notifies the user of an unresolved variable, and then waits for the user to type `import ` before triggering the suggestion. My first concern is that there are several different ways for an variable to be unresolved (e.g. simply mistyping the name.) I know that eclipse does some basic searching and will recommend a quick-fix to import possible libraries if you hover over the unresolved variable, so that might be an easy way to get a better metric as to whether or not the user actually wants to import something.

The fact that the user has to type `import ` before the suggestion triggers is also a little weird, it seems like the user would want to be notified they can auto-import things right when the annotation model detects that there is an unresolved variable that can be imported via a library.

Reply:
This is something that seems worth discussing and getting a consensus on.

I think we need to clearly define what exactly we want to suggest with this feature and when. We should chat about this tomorrow before / after the meeting.

Reply:
This functionality has been left as is. As per our meetings over the week, we want the focus to be on preventing the user from manually accomplishing something that can be automatically done, and not just as a reminder that they can do something automatically.